Core/Core/SwiftUIViews/ModuleBottomNavBar/ModuleBottomNavBarViewModel.swift

```diff
@@ -0,0 +1,54 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+class ModuleBottomNavBarViewModel: ObservableObject {
+    // MARK: - Dependencies
+
+    let router: Router
+    let hostingViewController: UIViewController?
+
```

Comment:
Please set weak here to avoid make retain cycle 

```
 private let router: Router
 weak private var hostingViewController: UIViewController?
```

Reply:
Good catch. Fixed!